clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/deploy/converters/DeployCloudFoundryServerGroupAtomicOperationConverter.java

```diff
@@ -0,0 +1,149 @@
+import com.netflix.spinnaker.clouddriver.cloudfoundry.artifacts.PackageArtifactCredentials;
+import com.netflix.spinnaker.clouddriver.cloudfoundry.client.CloudFoundryClient;
+import com.netflix.spinnaker.clouddriver.cloudfoundry.deploy.description.DeployCloudFoundryServerGroupDescription;
+import com.netflix.spinnaker.clouddriver.cloudfoundry.deploy.ops.DeployCloudFoundryServerGroupAtomicOperation;
+import com.netflix.spinnaker.clouddriver.cloudfoundry.provider.view.CloudFoundryClusterProvider;
+import com.netflix.spinnaker.clouddriver.cloudfoundry.security.CloudFoundryCredentials;
+import com.netflix.spinnaker.clouddriver.helpers.OperationPoller;
+import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperation;
+import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperations;
+import com.netflix.spinnaker.kork.artifacts.model.Artifact;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.util.List;
+import java.util.Map;
+
+@CloudFoundryOperation(AtomicOperations.CREATE_SERVER_GROUP)
+@Component
+public class DeployCloudFoundryServerGroupAtomicOperationConverter extends AbstractCloudFoundryServerGroupAtomicOperationConverter {
+  private final OperationPoller operationPoller;
+  private final ArtifactCredentialsRepository credentialsRepository;
+  private final CloudFoundryClusterProvider clusterProvider;
+
+  public DeployCloudFoundryServerGroupAtomicOperationConverter(@Qualifier("cloudFoundryOperationPoller") OperationPoller operationPoller,
+                                                               ArtifactCredentialsRepository credentialsRepository,
+                                                               CloudFoundryClusterProvider clusterProvider) {
+    this.operationPoller = operationPoller;
+    this.credentialsRepository = credentialsRepository;
+    this.clusterProvider = clusterProvider;
+  }
+
+  @Override
+  public AtomicOperation convertOperation(Map input) {
+    return new DeployCloudFoundryServerGroupAtomicOperation(operationPoller, convertDescription(input), clusterProvider);
+  }
+
+  @Override
+  public DeployCloudFoundryServerGroupDescription convertDescription(Map input) {
+    DeployCloudFoundryServerGroupDescription converted = getObjectMapper().convertValue(input, DeployCloudFoundryServerGroupDescription.class);
+    CloudFoundryCredentials credentials = getCredentialsObject(input.get("credentials").toString());
+    converted.setClient(credentials.getClient());
+    converted.setAccountName(credentials.getName());
+
+    converted.setSpace(findSpace(converted.getRegion(), converted.getClient())
+      .orElseThrow(() -> new IllegalArgumentException("Unable to find space '" + converted.getRegion() + "'.")));
+
+    Map artifactSource = (Map) input.get("artifactSource");
```

Comment:
Are these intended to be the canonical spinnaker artifacts?

Reply:
No. This is being replaced by the expected artifacts improvement in the end.